game-core/src/main/java/games/strategy/triplea/ui/PoliticalStateOverview.java

```diff
@@ -157,17 +157,16 @@ private JPanel getRelationshipLabel(final PlayerId player1, final PlayerId playe
         data.releaseReadLock();
       }
     }
-    final JComponent relationshipLabel = getRelationshipComponent(player1, player2, relType);
-    final JPanel relationshipLabelPanel = new JPanel();
-    relationshipLabelPanel.add(relationshipLabel);
-    relationshipLabelPanel.setBackground(getRelationshipTypeColor(relType));
-    return relationshipLabelPanel;
+    return getRelationshipComponent(player1, player2, relType, getRelationshipTypeColor(relType));
```

Comment:
Side-comment, this helper method is a bit ugly, in existing code. I'm seeing a "mixing of abstractions" code smell. Notably, above, we have very low level code, where we're getting locks on game data, computing a relationship type at very low level, then on this last line the abstraction goes to very high level "getRelationshipComponent".

To fix, we'd probably need to re-split the helper methods, so we coudl have stuff like:
```
RelationshipType relType = computeRelationship();
return createRelationshipComponent(label);
```
Sigh, no action needed on this comment, just noting that the existing code structure could use help.